src/reply.c

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2018-2021 Redis Labs Ltd. and Contributors
+ *
+ * This file is available under the Redis Labs Source Available License Agreement
+ */
+
+#include "reply.h"
+
+#include "redismodule.h"
+#include "tsdb.h"
+
+#include "rmutil/alloc.h"
+
+int ReplySeriesArrayPos(RedisModuleCtx *ctx,
+                        Series *s,
+                        bool withlabels,
+                        api_timestamp_t start_ts,
+                        api_timestamp_t end_ts,
+                        AggregationClass *aggObject,
+                        int64_t time_delta,
+                        long long maxResults,
+                        bool rev) {
+    RedisModule_ReplyWithArray(ctx, 3);
+    RedisModule_ReplyWithString(ctx, s->keyName);
+    if (withlabels) {
+        ReplyWithSeriesLabels(ctx, s);
+    } else {
+        RedisModule_ReplyWithArray(ctx, 0);
+    }
+    ReplySeriesRange(ctx, s, start_ts, end_ts, aggObject, time_delta, maxResults, rev);
+    return REDISMODULE_OK;
+}
+
+int ReplySeriesRange(RedisModuleCtx *ctx,
```

Comment:
@filipecosta90 this function is probably the only one that should be here, there's a lot of logic in this function that should be part of the command implementation and not here.
While it was part of module.c it made a bit of sense, but looking at this now its obvious we need to break this function.